File: app/src/main/java/com/thales/musicapp/musicsdk/utils/FileUtils.java
package com.thales.musicapp.musicsdk.utils;

import android.content.ContentResolver;
import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.provider.MediaStore;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

public    class FileUtils {

    public void getFiles(Context ctx){
        ContentResolver cr = ctx.getContentResolver();

        Uri uri = MediaStore.Audio.Media.EXTERNAL_CONTENT_URI;
        String selection = MediaStore.Audio.Media.IS_MUSIC + "!= 0";
        String sortOrder = MediaStore.Audio.Media.TITLE + " ASC";
        Cursor cur = cr.query(uri, null, selection, null, sortOrder);
        int count = 0;

        if(cur != null)
        {
            count = cur.getCount();

            if(count > 0)
            {
                while(cur.moveToNext())
                {
                    String data = cur.getString(cur.getColumnIndex(MediaStore.Audio.Media.DATA));
                    // Add code to get more column here

                    // Save to your list here
                }

            }

            cur.close();
        }
    }

    public static List<File> getAllAudios(Context c) {
        List<File> files = new ArrayList<>();
        String[] projection = { MediaStore.Audio.AudioColumns.DATA ,MediaStore.Audio.Media.DISPLAY_NAME};
        Cursor cursor = c.getContentResolver().query(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, projection, null, null, null);
        try {
            cursor.moveToFirst();
            do{
                files.add((new File(cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Audio.Media.DATA)))));
            }while(cursor.moveToNext());

            cursor.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return files;
    }
}

File: app/src/main/java/com/thales/musicsdk/musicsdk/utils/FileUtils.kt
package com.thales.musicsdk.musicsdk.utils

import android.content.Context
import android.provider.MediaStore
import com.thales.musicsdk.musicsdk.models.Song
import java.util.*

object FileUtils {

    /**
     * Method to get all the audio files from the device and adding the Name and file path to Data class
     */
    fun getAllAudios(c: Context): List<Song>? {
        val files: MutableList<Song> =
            ArrayList()
        val projection = arrayOf(
            MediaStore.Audio.AudioColumns.DATA,
            MediaStore.Audio.Media.DISPLAY_NAME
        )
        val cursor = c.contentResolver
            .query(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, projection, null, null, null)
        try {
            cursor!!.moveToFirst()
            do {
                files.add( Song(cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Audio.Media.DISPLAY_NAME)),
                cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Audio.Media.DATA)
                )))
            } while (cursor.moveToNext())
            cursor.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return files
    }

    /**
     * Method to remove the file extension from file name.
     */
    fun getFileName(fileName: String): String? {
        if (fileName.indexOf(".") > 0)
            return fileName?.substring(0, fileName?.lastIndexOf("."));
        else return ""
    }
}